src/Google/Ads/GoogleAds/Lib/V7/GoogleAdsClientBuilder.php

```diff
@@ -290,11 +324,41 @@ public function validate()
             );
         }
 
+        if (
+            !empty($this->transport) && $this->transport === 'grpc'
+        ) {
+            self::validateGrpcSupport();
+        }
+
         if (is_null($this->logLevel)) {
             $this->logLevel = LogLevel::INFO;
         } elseif (!defined('Psr\Log\LogLevel::' . strtoupper($this->logLevel))) {
             throw new InvalidArgumentException("The log level must be a valid PSR log level");
         }
+
+        if ($this->grpcChannelIsInsecure && $this->grpcChannelCredential !== null) {
+            throw new InvalidArgumentException(
+                'The gRPC channel credential can only be set when the gRPC channel is set as ' .
+                'not insecure.'
```

Comment:
Can we rename the variable to the "grpcChannelIsSecure" and default it to "true"?
It's error-prone and hard to maintain int the future when using the double negation like this.